runtime/pkg/credentials/credentials.go

```diff
@@ -0,0 +1,52 @@
+package credentials
```

Comment:
I think the package name/path should indicate that it's AWS related. Just `runtime/pkg/credentials` sounds like it's related to Rill authentication.

Also, maybe this logic just best belongs in a file in `runtime/drivers/s3` for now?